Bottom: 31afbc3b578d3df9f9bb440bd0defa44c47b6a4d
Top:    923a59f751ef5a438528586054bc60ead4a003fc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 17:14:04 +0000

hid/gtk: Fix NetlistChanged action crash when exporting from the commandline.

I broke this in commit 7308f512307158944482227d58e66373fd023d62 when I
added a missing notification to the GUI when free'ing the netlist. It turns
out that a command-line invoked export (such as used to export images in the
doc/ build) will hit this code-path, and provoke the GUI netlist window code
to try and update before the GUI is actually loaded.


---

diff --git a/src/hid/gtk/gui-netlist-window.c b/src/hid/gtk/gui-netlist-window.c
index 447b10e..922a03c 100644
--- a/src/hid/gtk/gui-netlist-window.c
+++ b/src/hid/gtk/gui-netlist-window.c
@@ -1009,6 +1009,9 @@ ghid_netlist_window_update (gboolean init_nodes)
 static gint
 GhidNetlistChanged (int argc, char **argv, Coord x, Coord y)
 {
+  if (!ghidgui->is_up)
+    return 0;
+
   loading_new_netlist = TRUE;
   ghid_netlist_window_update (TRUE);
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (disable_all_button),
